core/src/main/java/org/apache/calcite/rel/rules/ProjectJoinTransposeRule.java

```diff
@@ -38,7 +39,7 @@
  */
 public class ProjectJoinTransposeRule extends RelOptRule {
   public static final ProjectJoinTransposeRule INSTANCE =
-      new ProjectJoinTransposeRule(expr -> true,
+      new ProjectJoinTransposeRule(expr -> !(expr instanceof RexOver),
           RelFactories.LOGICAL_BUILDER);
 
```

Comment:
Not that `ProjectJoinTransposeRule#preserveExprCondition` means expressions that should be preserved in the project after transposing instead of in the origin project. So `expr -> !(expr instanceof RexOver)` means `RexOver` should be left in the origin project.

Reply:
Thanks for the clarification!

But how about `ProjectCorrelateTransposeRule`? I haven't test for if that one causes the same issue too.

Reply:
Now I am testing all `ProjectXXXTransposeRule`. But I prefer to open another JIRA if I find some unexpected behavior. What do you think of it?

Reply:
I am with you. Also I just opened [CALCITE-3004](https://issues.apache.org/jira/browse/CALCITE-3004), the reason seems to be not related.

Reply:
I also find it. May I take it?

Reply:
Sure. Thank you very much!